From f00d5546c9e989dd68ce0de0190cd0e043b0f1f5 Mon Sep 17 00:00:00 2001
From: Arjan Opmeer <arjan.opmeer@gmail.com>
Date: Tue, 9 Jul 2024 13:55:41 +0200
Subject: [PATCH] tools/btmgmt: Fix --index option for non-interactive mode

In non-interactive mode the --index option does not work because the
call to mgmt_set_index() is made after bt_shell_attach().

Fixes: https://github.com/bluez/bluez/issues/893

Upstream-Status: Backport [https://github.com/bluez/bluez/commit/f00d5546c9e989dd68ce0de0190cd0e043b0f1f5]
---
 tools/btmgmt.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/btmgmt.c b/tools/btmgmt.c
index 9b7f851bd8..436c2bb21f 100644
--- a/tools/btmgmt.c
+++ b/tools/btmgmt.c
@@ -51,8 +51,8 @@ int main(int argc, char *argv[])
 		return EXIT_FAILURE;
 	}
 
-	bt_shell_attach(fileno(stdin));
 	mgmt_set_index(index_option);
+	bt_shell_attach(fileno(stdin));
 	status = bt_shell_run();
 
 	mgmt_remove_submenu();
